cmd/processor/app/processor.go

```diff
@@ -649,3 +657,49 @@ func (p *Processor) setWorkersStatus(triggerInstance trigger.Trigger, status sta
 
 	return nil
 }
+
+// creates a signal handler, so on signal processor gracefully terminated
+func (p *Processor) handleSignals() {
+	var captureSignal = make(chan os.Signal, 1)
+	signal.Notify(captureSignal, syscall.SIGTERM, syscall.SIGABRT, syscall.SIGINT)
+	p.actOnSignal(<-captureSignal)
+}
+
+func (p *Processor) actOnSignal(signal os.Signal) {
+	p.logger.WarnWith("", "Got system signal", signal.String())
+	p.terminateAllTriggers()
+	os.Exit(0)
+}
+
+func (p *Processor) terminateAllTriggers() {
+	// counter to control trigger termination
+	gracefulShutdownTimeout := time.NewTicker(defaultGracefulShutdownTimeout)
+	tasksCounter := int32(0)
+
+	for _, triggerInstance := range p.triggers {
+
+		// adding new task
+		atomic.AddInt32(&tasksCounter, 1)
+
+		// goroutine to drain trigger, decrements counter when it's done
+		go func(triggerInstance trigger.Trigger, counter *int32) {
+			defer atomic.AddInt32(counter, -1)
+			triggerInstance.SignalWorkerDraining()
+		}(triggerInstance, &tasksCounter)
+	}
+
+	tasksCounterCheckTicker := time.NewTicker(time.Second)
```

Comment:
why not simply wait group and wait? if it takes more than 30 seconds, the signaler will restart you anyway